lib/midcom/helper/imagepopup/handler/upload.php

```diff
@@ -0,0 +1,159 @@
+        }
+        
+        // Get the data
+        $temp_name = $temp['tmp_name'];
+        $mimetype = $temp['type'];
+        $parentguid = "";
+        
+        // Do we have guid ?
+        switch($handler_id)
+        {
+            case '____ais-imagepopup-upload_image_noobject':
+                $parentguid = $this->_topic->guid;
+                break;
+            case '____ais-imagepopup-upload_image':
+                $parentguid = $args[0];
+                break;
+        }
+        
+        // Set modified filename
+        $filename = $this->get_modify_filename($temp['name']);
+        
+        // Insert the image into database
+        $attachment = $this->insert_database($filename, $mimetype, $parentguid);
+        $data['attachment'] = $attachment;
+        
+        // Get target to write the file
+        $target = $this->get_data_from_database($filename, $parentguid);
+        
+        // Write the file
+        $this->write_the_file($temp_name, $target);
+        
+        // Make a response for editor.uploadImages() function
+        $location = midcom_db_attachment::get_url($attachment);
+        $data['location'] = $location;
+        
+        // Return image location as JSON
+        return new midcom_response_json(array('location' => $location));
+    }
+    
+    private function get_modify_filename($filename)
+    {
+        $filename = midcom_db_attachment::safe_filename($filename);
+        $pieces = explode('.', $filename);
+        $core = "modified";
+        $extension = array_pop($pieces);
+        $time = $this->get_current_time('Y-m-d-H-i-s');
+        $modifyFilename = $core . $time . "." . $extension;
+        
+        return $modifyFilename;
+    }
+    
+    private function get_current_time($pattern)
+    {
+        return date ($pattern, time());
+    }
+    
+    private function get_data_from_database($filename, $parentguid)
+    {
+        $filename = midcom_db_attachment::safe_filename($filename);
+        $query = midcom_db_attachment::new_query_builder();
+        $query->add_constraint('name', '=', $filename);
+        $query->add_constraint('parentguid', '=', $parentguid);
+        $entry = $query->execute();
+        
+        if(count($entry) == 1)
+        {
+            return $entry[0];
+        } 
+        else if(count($entry) > 1)
+        {
+            throw new midcom_error('There is more than just one object' . midcom_connection::get_error_string());
+        }
+    }
+    
+    private function write_the_file($tmp, $target)
+    {
+        $source = fopen($tmp, 'r');
+        if(!$source)
+        {
+            throw new midcom_error("Could not open file " . $tmp . " for reading: " . midcom_connection::get_error_string());
+        }
+        $stat = $target->copy_from_handle($source);
+        if(!$stat)
+        {
+            throw new midcom_error('Failed to copy from handle: ' . midcom_connection::get_error_string());
+        }
+        fclose($source);
```

Comment:
`fclose` should be called before `throw`, otherwise, the file handle will stay open